src/multivariate/solvers/second_order/newton.jl

```diff
@@ -40,51 +44,55 @@ end
 
 function initial_state(method::Newton, options, d, initial_x)
     T = eltype(initial_x)
-    n = length(initial_x)
-    # Maintain current gradient in gr
-    s = similar(initial_x)
 
     value_gradient!!(d, initial_x)
     hessian!!(d, initial_x)
 
     NewtonState(
-        copy(initial_x), # Maintain current state in state.x
-        copy(initial_x), # Maintain previous state in state.x_previous
-        T(NaN), # Store previous f in state.f_x_previous
-        Cholesky(similar(d.H, T, 0, 0), :U, BLAS.BlasInt(0)),
-        similar(initial_x), # Maintain current search direction in state.s
+        copy(initial_x),                  # x
+        copy(initial_x),                  # x_previous
+        T(NaN),                           # f_x_previous
+        Cholesky(similar(d.H, T, 0, 0),   # F
+                :U, 0),
+        similar(initial_x),               # s
         @initial_linesearch()...,
     )
 end
 
-function update_state!(d, state::NewtonState, method::Newton)
-    # Search direction is always the negative gradient divided by
-    # a matrix encoding the absolute values of the curvatures
-    # represented by H. It deviates from the usual "add a scaled
-    # identity matrix" version of the modified Newton method. More
-    # information can be found in the discussion at issue #153.
+# Default solver that handles common matrix types intelligently
+function default_newton_solve!(d, state::NewtonState, method::Newton)
+    H = NLSolversBase.hessian(d)
+    g = gradient(d)
     T = eltype(state.x)
 
-    if typeof(NLSolversBase.hessian(d)) <: AbstractSparseMatrix
-        state.s .= .-(NLSolversBase.hessian(d) \ convert(Vector{T}, gradient(d)))
+    if H isa AbstractSparseMatrix
+        state.s .= -(H \ convert(Vector{T}, gradient(d)))
     else
-        state.F = cholesky!(Positive, NLSolversBase.hessian(d))
-        if typeof(gradient(d)) <: Array
-            # is this actually StridedArray?
-            ldiv!(state.s, state.F, -gradient(d))
-        else
-            # not Array, we can't do inplace ldiv
-            gv = Vector{T}(undef, length(gradient(d)))
-            copyto!(gv, -gradient(d))
+        # Use PositiveFactorizations for robustness on dense matrices
+         # Search direction is always the negative gradient divided by
+         # a matrix encoding the absolute values of the curvatures
+         # represented by H. It deviates from the usual "add a scaled
+         # identity matrix" version of the modified Newton method. More
+         # information can be found in the discussion at issue #153.
+         state.F = cholesky!(Positive, H)
+         if g isa Array
+            ldiv!(state.s, state.F, -g)
```

Comment:
Use
```suggestion
            ldiv!(state.s, state.F, g)
            state.s .= .-state.s
```
to reduce allocations?

Reply:
From the original code it looks like the gradient was negated?

```
ldiv!(state.s, state.F, -gradient(d))
```

Otherwise yes to the suggestion for reduced allocations.